src/crt0.s

```diff
@@ -124,3 +130,58 @@ IntrMain_RetAddr:
 	.pool
 
 	.align 2, 0 @ Don't pad with nop.
+
+@ Fills initialized IWRAM and EWRAM sections in RAM from LMA areas in ROM
+InitializeWorkingMemory:
+	push {r0-r3,lr}
+	ldr r0, =__iwram_lma
+	ldr r1, =__iwram_start
+	ldr r2, =__iwram_end
+	cmp r1, r2
+	beq skip_iwram_copy
+	bl CopyMemory_DMA
+skip_iwram_copy:
+	ldr r0, =__ewram_lma
+	ldr r1, =__ewram_start
+	ldr r2, =__ewram_end
+	cmp r1, r2
+	beq skip_ewram_copy
+	bl CopyMemory_DMA
+skip_ewram_copy:
+	pop {r0-r3,lr}
+	bx lr
+
+@ Uses a DMA transfer to load from r0 into r1 until r2
+CopyMemory_DMA:
+	subs r2, r2, r1
+	lsr r2, r2, #2
+	ldr r3, =0x040000D4
+	str r0, [r3]
+	ldr r3, =0x040000D8
+	str r1, [r3]
+	ldr r3, =0x040000DC
+	mov r4, #0x80000000
+	orr r4, r4, #(1 << 26)
+	orr r2, r2, r4
+	str r2, [r3]
```

Comment:
I think you could say `ldr r3, =REG_DMA3SAD`, etc (these are defined in `constants/gba_constants.inc`).

Also, a pretty neat thing—`DMA3SAD`, `DMA3DAD`, and `DMA3CNT` are adjacent, which means they can all be written to with a single `stmia`, e.g.
```suggestion
	mov r4, #0x80000000
	orr r4, r4, #(1 << 26)
	orr r2, r2, r4
	ldr r3, =DMA3
	stmia r3, {r0, r1, r2}
```

Reply:
That is a neat thing, implemented and tested! Thank you 😄